packages/devtools_app_shared/lib/src/service/isolate_manager.dart

```diff
@@ -129,6 +136,11 @@ final class IsolateManager with DisposerMixin {
 
       if (_selectedIsolate.value == null) {
         _setSelectedIsolate(event.isolate);
+      } else if (_shouldReselectMainIsolate) {
```

Comment:
I think this case should be handled first. In theory, a hot restart could result in a couple of isolates coming up.

If we're already assuming that the first isolate to restart is the main isolate and VM developer mode is disabled, `_selectedIsolate.value` will be null and we'll execute the first case. When the second isolate comes up, `_shouldReselectMainIsolate` will still be `true`, so we'll select the second isolate which, based on our assumptions, isn't the main isolate.

Reply:
We may also want to check the name of the isolate. @bkonyi is it safe to assume the main isolate will always have a name of 'main'? This is what I have seen while debugging but I'm not sure if my local testing covers all the possible cases.

Reply:
There's no guarantees provided around isolate names as they're completely dependent on the embedder. However, I'd be surprised if the standalone VM or the Flutter engine deviated from using `main` for the first isolate spawned that's running user code. 

Flutter might have scenarios where there's multiple "main" isolates though, assuming that it's possible for multiple rendering surfaces to have their own UI isolates, but I'm not familiar enough with this area to know.

Reply:
I'd be ok with us assuming for now and adjusting in the future if it turns out there are cases that it misses :)